src/Tests/Directory.Build.targets

```diff
@@ -26,6 +27,47 @@
     
   </ItemGroup>
 
+  <Target Name="LayoutSdk">
+    <MSBuild Projects="$(RepoRoot)\src\LayoutSdk\LayoutSdk.proj"
+             Targets="LayoutSdk"/>
+  </Target>
+
+  <Target Name="TestAsTool" DependsOnTargets="Pack;LayoutSdk;_InnerGetTestsToRun">
+    <PropertyGroup>
+      <TestLocalToolFolder>$(ArtifactsTmpDir)$(ToolCommandName)\</TestLocalToolFolder>
+    </PropertyGroup>
+
+    <RemoveDir Directories="$(TestLocalToolFolder)" />
+    <MakeDir Directories="$(TestLocalToolFolder)" />
+
+    <MakeDir Directories="$(ArtifactsTestResultsDir)" />
+
+    <!-- Remove tool package from NuGet cache, so that local changes will be reflected if the package version
+         hasn't changed. -->
+    <RemoveDir Directories="$(NuGetPackageRoot)$(ToolCommandName)\$(PackageVersion)" />
+
+    <Exec Command="dotnet new tool-manifest"
+      WorkingDirectory="$(TestLocalToolFolder)"/>
+    
+    <Exec Command="dotnet tool install --local $(ToolCommandName) --version $(PackageVersion) --add-source $(ArtifactsShippingPackagesDir)"
+      WorkingDirectory="$(TestLocalToolFolder)"/>
+
+    <PropertyGroup>
+      <ResultsXmlPath>$(ArtifactsTestResultsDir)$(_ResultFileNameNoExt).xml</ResultsXmlPath>
```

Comment:
Doesn't arcade define these already? Should we reuse it from there if that's true?